shared/actions/chat/index.js

```diff
@@ -1016,6 +1016,33 @@ function* _exitSearch() {
   yield put(Creators.setInboxFilter([]))
 }
 
+function* _selectSearchResultWithMovement({payload: {movement}}: Constants.SelectSearchResultIdWithMovement) {
+  const searchResults = yield select(({chat: {searchResults}}: TypedState) => searchResults)
+  const selectedSearchId = yield select(({chat: {selectedSearchId}}: TypedState) => selectedSearchId)
+  const index = searchResults.indexOf(selectedSearchId)
+
+  const nextIndex = movement === 'down'
+    ? Math.min(index + 1, searchResults.count() - 1)
+    : Math.max(index - 1, 0)
+  const nextSelectedSearchId = searchResults.get(nextIndex)
+  yield put(Creators.selectSearchResultId(nextSelectedSearchId))
+}
+
+function* _addSelectedSearchResult() {
+  const selectedSearchId = yield select(({chat: {selectedSearchId}}: TypedState) => selectedSearchId)
+  yield put(Creators.stageUserForSearch(selectedSearchId))
+  yield put(Creators.clearSearchResults())
+}
+
+function* _selectFirstSearchResult() {
+  const searchResults = yield select(({chat: {searchResults}}: TypedState) => searchResults)
+  yield put(Creators.selectSearchResultId(searchResults.first()))
+}
+
+function* _clearSelectedSearchResult() {
+  yield put(Creators.selectSearchResultId(null))
```

Comment:
no this lives solely in the chat store. This isn't an entity, it's like the list of what search result ids to show. It's some state that affects the ui.